tests/test_dice_interface/test_dice_genetic.py

```diff
@@ -177,7 +177,43 @@ def test_predict_custom(self, desired_class, sample_custom_query_2, mocker):
         mocker.patch('dice_ml.model_interfaces.base_model.BaseModel.get_output', return_value=[[0, 0.5, 0.5]])
         custom_preds = self.exp._predict_fn_custom(sample_custom_query_2, desired_class)
         assert custom_preds[0] == desired_class
-
+    
+    # Testing if the shapes of the predictions are correct for multiclass classification
+    @pytest.mark.parametrize(("desired_class", "method"), [(1, "genetic")])
+    def test_multiclass_nn(self, desired_class, method):
+        pytest.importorskip("torch")
+        pytest.importorskip("sklearn")
+
+        backend = "PYT"
+        dataset = helpers.load_custom_testing_dataset_multiclass()
+
+        # Transform the categorical data to number to test the neural network
+        import sklearn 
+
+        label_enc = sklearn.preprocessing.LabelEncoder()
+        dataset['Categorical'] = label_enc.fit_transform(dataset['Categorical'])
+
+        d = dice_ml.Data(dataframe=dataset, continuous_features=['Numerical', 'Categorical'], outcome_name='Outcome')
+
+        # Load the neural network for multiclass classification and generate an explainer
+        df = d.data_df
+        num_class = len(df['Outcome'].unique())
+        model = MulticlassNetwork(input_size=df.drop("Outcome", axis=1).shape[1], num_class=num_class)
+        m = dice_ml.Model(model=model, backend=backend)
+        exp = dice_ml.Dice(d, m, method=method)
+
+        try:
+            # Test the function that returns the predictions
+            _, _, preds = exp.build_KD_tree(
+                df.copy(), desired_range=None, desired_class=desired_class, 
+                predicted_outcome_name=d.outcome_name + '_pred'
+            )
+            assert hasattr(preds, "shape"), "The returned object that contains the predictions doesn't have a 'shape' attribute."
+            assert preds.shape[0] == df.shape[0], "The number of predictions differ from the number of elements in the passed dataset."
+        except ValueError as ve:
```

Comment:
Why do we need this try and except in a test case? If the test is suppose to catch the actual regressions and we can just keep the code under try and let the executor of the test determine the cause of the error.